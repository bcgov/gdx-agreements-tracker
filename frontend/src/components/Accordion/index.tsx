import * as React from "react";
import { Accordion as MUIAccordion } from "@mui/material/";
import AccordionSummary from "@mui/material/AccordionSummary";
import AccordionDetails from "@mui/material/AccordionDetails";
import Typography from "@mui/material/Typography";
import ExpandMoreIcon from "@mui/icons-material/ExpandMore";
import { styled } from "@mui/material";
import bcgovTheme from "../../bcgovTheme";

export const Accordion = ({
  children,
  sectionTitle,
}: {
  children: JSX.Element | JSX.Element[];
  sectionTitle: string;
}) => {
  const StyledAccordionSummary = styled(AccordionSummary)({
    backgroundColor: "#f3f3f3",
  });

  const StyledTypography = styled(Typography)({
    color: bcgovTheme.palette.primary.main,
    fontWeight: "bold",
  });

  return (
    <MUIAccordion defaultExpanded>
      <StyledAccordionSummary
        expandIcon={<ExpandMoreIcon />}
        aria-controls="panel-content"
        aria-label="button"
      >
        <StyledTypography>{sectionTitle}</StyledTypography>
      </StyledAccordionSummary>
      <AccordionDetails>{children}</AccordionDetails>
    </MUIAccordion>
  );
};
