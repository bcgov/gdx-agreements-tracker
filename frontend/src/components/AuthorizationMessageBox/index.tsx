import { FormHelperText } from "@mui/material";

export const AuthorizationMessageBox = ({ message }: { message: string }) => {
  return <FormHelperText>{message}</FormHelperText>;
};
