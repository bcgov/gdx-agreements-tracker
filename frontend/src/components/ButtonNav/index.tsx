import React from "react";
import { Link, useMatch, useResolvedPath } from "react-router-dom";
import type { LinkProps } from "react-router-dom";

/**
 * Returns react router link with active class if matches route.
 *
 * @param {LinkProps} root0 Link properties.
 * @param {ReactNode | undefined} root0.children Child nodes.
 * @param {To} root0.to Destination.
 * @returns {React.ReactNode}
 */
export const ButtonNav = ({ children, to, ...props }: LinkProps) => {
  const resolved = useResolvedPath(to);
  const match = useMatch({ path: resolved.pathname, end: true });
  if (match) {
    props.className = `${props.className ?? ""} active`;
  }
  return (
    <Link to={to} {...props}>
      {children}
    </Link>
  );
};

export default ButtonNav;
