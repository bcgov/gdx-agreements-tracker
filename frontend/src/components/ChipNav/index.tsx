import * as React from "react";
import { styled } from "@mui/material/styles";
import Chip from "@mui/material/Chip";
import { IChipNav } from "../../types";
import { Box, Paper } from "@mui/material";
import { NavLink } from "react-router-dom";
import bcgovTheme from "../../bcgovTheme";

export const ChipNav = ({
  navLinks,
  navLinksRight = [],
}: {
  navLinks: IChipNav[];
  navLinksRight?: IChipNav[];
}) => {
  const StyledListItem = styled("li")(({ theme }) => ({
    margin: theme.spacing(0.5),
  }));

  const StyledNavLink = styled(NavLink)(() => ({
    textDecoration: "none",
  }));

  /**
   * Renders a single ChipNav link.
   *
   * @param   {IChipNav}    link The ChipNav link to render.
   * @returns {JSX.Element}
   */
  const renderNavLink = (link: IChipNav): JSX.Element => {
    return (
      <StyledListItem key={link.key}>
        <StyledNavLink to={link.url} end>
          {({ isActive }) => (
            <Chip
              sx={{
                backgroundColor: isActive ? "#D3D3D3" : bcgovTheme.palette.primary.main,
                color: isActive ? "#000" : "#FFF",
              }}
              label={link.name}
            />
          )}
        </StyledNavLink>
      </StyledListItem>
    );
  };

  return (
    <Paper
      elevation={3}
      sx={{
        display: "flex",
        justifyContent: "space-between",
        flexWrap: "wrap",
        p: 0.5,
        mb: 2,
        mt: 0,
      }}
    >
      <Box sx={{ display: "flex", listStyle: "none" }}>{navLinks.map(renderNavLink)}</Box>
      {navLinksRight.length > 0 && (
        <Box sx={{ display: "flex", listStyle: "none" }}>{navLinksRight.map(renderNavLink)}</Box>
      )}
    </Paper>
  );
};
