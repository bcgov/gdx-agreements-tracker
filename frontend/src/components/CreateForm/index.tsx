import React from "react";
import { Box, Button } from "@mui/material";
import { Form, Formik, FormikHelpers, FormikValues } from "formik";
import { FormLayout } from "../GDXForm/FormLayout";
import { FormInput } from "../FormInput";
import { IEditFields } from "../../types";

/**
 * Destructured as:
 * @param    {​​​​​​Object[]}​​​​​​ initialValues - Contains all the initial values for Formik to use
 * @param    {​​​​​​​​​​​​​​​​​​​​string}​​​​​​​​​​​​​​​​​​​​ onSubmit - Handles the submit functionality for the form
 * @param    {Object[]}​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​ editFields - The field properties used to render the fields for an edit form
 * @returns  {React.JSX}
 */

export const CreateForm = ({
  initialValues,
  onSubmit,
  editFields,
}: {
  initialValues: FormikValues;
  // todo: Define a good type. "Any" type temporarily permitted.
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  onSubmit: ((values: unknown, formikHelpers: FormikHelpers<any>) => void | Promise<any>) &
    Function;
  editFields: IEditFields[];
}) => {
  return (
    <Formik initialValues={initialValues} onSubmit={onSubmit}>
      {({ setFieldValue, values, handleChange, dirty }) => {
        return (
          <Form>
            <FormLayout>
              {editFields.map(({ fieldName, fieldType, fieldLabel, width, tableName }) => {
                return (
                  <FormInput
                    setFieldValue={setFieldValue}
                    fieldValue={values?.[fieldName]}
                    fieldName={fieldName}
                    fieldType={fieldType}
                    fieldLabel={fieldLabel}
                    handleChange={handleChange}
                    width={width}
                    key={fieldName}
                    tableName={tableName}
                  />
                );
              })}
            </FormLayout>
            <Box
              m={1}
              display="flex"
              justifyContent="flex-end"
              alignItems="flex-end"
              role={"submit_button"}
            >
              <Button
                type="submit"
                variant="contained"
                color="success"
                disabled={dirty ? false : true}
              >
                Submit
              </Button>
            </Box>
          </Form>
        );
      }}
    </Formik>
  );
};
