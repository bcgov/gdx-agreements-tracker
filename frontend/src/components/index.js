export { ButtonNav } from "./ButtonNav";
export { Debug } from "./Debug";
export { Loader } from "./Loader";
export { Main } from "./Layout/Main";
export { PageLink } from "./PageLink";
export { RowEditDeleteChip } from "./RowEditDeleteChip";
export { Sidebar } from "./Sidebar";
export { DetailsList } from "./DetailsList";
export { Table } from "./Table";
