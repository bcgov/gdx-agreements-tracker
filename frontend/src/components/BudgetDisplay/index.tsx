import { Box, Grid } from "@mui/material";
import { GDXList } from "components/GDXList";
import { Loader } from "components/Loader";
import { useAxios } from "hooks/useAxios";
import React from "react";
import { useQuery } from "react-query";

export const BudgetDisplay = ({ apiUrl }: { apiUrl: string }) => {
  const { axiosAll } = useAxios();

  const getBudgetData = async () => {
    const budget = await axiosAll()
      .get(apiUrl)
      .then((budgetData: any) => {
        return budgetData;
      });
    return budget.data.data;
  };

  // Queries
  const { data, isLoading } = useQuery("contract budget", getBudgetData, {
    // todo: When there is an edit and view form built, reassess these options.
    refetchOnWindowFocus: false,
    retryOnMount: false,
    refetchOnReconnect: false,
    retry: false,
    staleTime: Infinity,
  });

  return (
    <Grid container spacing={2}>
      {!data ? (
        <Loader />
      ) : (
        data.map((budget: any) => {
          return <GDXList data={budget} title={`Fiscal ${budget.fiscal_year}`} />;
        })
      )}
    </Grid>
  );
};