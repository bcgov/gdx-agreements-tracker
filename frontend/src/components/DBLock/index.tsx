import { Grid, Skeleton } from "@mui/material";
import LockPersonIcon from "@mui/icons-material/LockPerson";
import { FormEditButton } from "components/FormEditButton";
import { useParams } from "react-router";

interface IDBLockProps {
  handleDbLock: Function;
  removeLock: Function;
  // todo: Define a good type. "Any" type temporarily permitted.  The query:any type will be fixed when we upgrade to ReactQuery V4.
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  query: any;
  handleFormType: (formType: "read" | "edit" | "new") => void;
}

/* This is a functional component called `DBLock` that takes in four props: `handleDbLock`,
`removeLock`, `query`, and `handleEditMode`. It returns a JSX element that renders a grid with
various components, including `Skeleton`, `LockPersonIcon`, and `FormEditButton`. The `query` prop
is of type `any` for now, but will be fixed when the code is upgraded to ReactQuery V4. The
component also uses the `useParams` hook from React Router to get the `projectId` parameter from the
URL. When the `FormEditButton` is clicked, it calls various functions passed in as props to remove a
lock on a database row, handle a new lock, and set the edit mode to true. */

export const DBLock = ({ handleDbLock, removeLock, query, handleFormType }: IDBLockProps) => {
  const { projectId } = useParams();

  return (
    <Grid container spacing={2}>
      <Grid item xs={6} md={6}>
        <Skeleton variant="rectangular" />
      </Grid>
      <Grid item xs={6} md={6}>
        <Skeleton variant="rectangular" />
      </Grid>
      <Grid item xs={6} md={6}>
        <LockPersonIcon />
        {/* <h1>section locked for editing by: {query?.data?.dbRowLock.locked_by}</h1> */}
      </Grid>
      <Grid item xs={6} md={6}></Grid>
      <Grid item xs={4} md={4}></Grid>
      <Grid item xs={4} md={4}></Grid>
      <Grid item xs={4} md={4}>
        <FormEditButton
          buttonText="Take Over Editing"
          onClick={async () => {
            await removeLock(query?.data?.dbRowLock).then(async () => {
              await handleDbLock(query, projectId).then(async () => {
                await query.refetch().then(() => {
                  handleFormType("edit");
                });
              });
            });
          }}
        />
      </Grid>
    </Grid>
  );
};
