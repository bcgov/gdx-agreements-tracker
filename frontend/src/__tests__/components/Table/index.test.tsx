import React from "react";
import { mount, shallow } from "enzyme";
import { Table } from "components";
import { GridColDef } from "@mui/x-data-grid";
import { IconButton } from "@mui/material";
import RemoveRedEyeIcon from "@mui/icons-material/RemoveRedEye";

const mockFunction = jest.fn();

const columns: GridColDef[] = [
  { field: "id", headerName: "ID", width: 90 },
  {
    field: "edit",
    headerName: "edit",
    width: 60,
    editable: true,
    renderCell: () => {
      return (
        <IconButton className="mockButton" onClick={mockFunction}>
          <RemoveRedEyeIcon />
        </IconButton>
      );
    },
  },
];

const rows = [{ id: 1 },{ id: 2 },{ id: 3 }]

describe("Tests different renders of the <Renderer /> component", () => {
  //This function allows you to render the component with different isLoading props to test different outcomes
  const wrapper = ({ loading, allowEdit }: { loading: boolean; allowEdit?: boolean }) => {
    return mount(
      <Table
        rows={rows}
        columns={columns}
        loading={loading}
        allowEdit={allowEdit}
      />
    );
  };

  it("test 1", () => {
    expect(wrapper({ loading: false, allowEdit: true }).html()).toMatchSnapshot();
  });

  it("test 2", () => {
    wrapper({ loading: false, allowEdit: true }).find(".mockButton").hostNodes().at(0).simulate("click");
    expect(mockFunction.mock.calls.length).toEqual(1);
  });
});
