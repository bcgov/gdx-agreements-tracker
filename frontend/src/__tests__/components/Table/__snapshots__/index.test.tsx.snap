// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests different renders of the <Renderer /> component test 1 1`] = `"<div class=\\"MuiBox-root css-jgwz4o\\"><div class=\\"MuiDataGrid-root MuiDataGrid-autoHeight MuiDataGrid-root--densityStandard css-1y321ta-MuiDataGrid-root\\" role=\\"grid\\" aria-colcount=\\"2\\" aria-rowcount=\\"4\\" aria-multiselectable=\\"false\\"><div><div><div class=\\"MuiBox-root css-vulnvf\\"><div class=\\"MuiFormControl-root MuiTextField-root css-3be3ve-MuiFormControl-root-MuiTextField-root-MuiDataGrid-toolbarQuickFilter\\" aria-label=\\"Search\\"><div class=\\"MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedStart MuiInputBase-adornedEnd css-1ptx2yq-MuiInputBase-root-MuiInput-root\\"><svg class=\\"MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root\\" focusable=\\"false\\" aria-hidden=\\"true\\" viewBox=\\"0 0 24 24\\" data-testid=\\"SearchIcon\\"><path d=\\"M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z\\"></path></svg><input aria-invalid=\\"false\\" placeholder=\\"Searchâ€¦\\" type=\\"search\\" class=\\"MuiInputBase-input MuiInput-input MuiInputBase-inputTypeSearch MuiInputBase-inputAdornedStart MuiInputBase-inputAdornedEnd css-c63i49-MuiInputBase-input-MuiInput-input\\" value=\\"\\" id=\\"mui-1\\"><button class=\\"MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-fp9tcu-MuiButtonBase-root-MuiIconButton-root\\" tabindex=\\"0\\" type=\\"button\\" aria-label=\\"Clear\\"><svg class=\\"MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root\\" focusable=\\"false\\" aria-hidden=\\"true\\" viewBox=\\"0 0 24 24\\" data-testid=\\"CloseIcon\\"><path d=\\"M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z\\"></path></svg><span class=\\"MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root\\"></span></button></div></div></div></div></div><div class=\\"MuiDataGrid-main css-204u17-MuiDataGrid-main\\"><div class=\\"MuiDataGrid-columnHeaders css-f3jnds-MuiDataGrid-columnHeaders\\" style=\\"min-height: 56px; max-height: 56px; line-height: 56px;\\"><div class=\\"MuiDataGrid-columnHeadersInner MuiDataGrid-columnHeadersInner--scrollable css-1s0hp0k-MuiDataGrid-columnHeadersInner\\" aria-rowindex=\\"1\\" role=\\"row\\" style=\\"transform: translate3d(0px, 0px, 0px);\\"><div class=\\"MuiDataGrid-columnHeader MuiDataGrid-columnHeader--sortable\\" data-field=\\"id\\" style=\\"width: 90px; min-width: 90px; max-width: 90px;\\" role=\\"columnheader\\" tabindex=\\"0\\" aria-colindex=\\"1\\" aria-sort=\\"none\\"><div class=\\"MuiDataGrid-columnHeaderDraggableContainer\\" draggable=\\"false\\"><div class=\\"MuiDataGrid-columnHeaderTitleContainer\\"><div class=\\"MuiDataGrid-columnHeaderTitleContainerContent\\"><div class=\\"MuiDataGrid-columnHeaderTitle css-1jbbcbn-MuiDataGrid-columnHeaderTitle\\" aria-label=\\"\\" data-mui-internal-clone-element=\\"true\\">ID</div></div><div class=\\"MuiDataGrid-iconButtonContainer css-ltf0zy-MuiDataGrid-iconButtonContainer\\"><button class=\\"MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-1pe4mpk-MuiButtonBase-root-MuiIconButton-root\\" tabindex=\\"-1\\" type=\\"button\\" aria-label=\\"Sort\\" title=\\"Sort\\"><svg class=\\"MuiSvgIcon-root MuiSvgIcon-fontSizeSmall MuiDataGrid-sortIcon css-ptiqhd-MuiSvgIcon-root\\" focusable=\\"false\\" aria-hidden=\\"true\\" viewBox=\\"0 0 24 24\\" data-testid=\\"ArrowUpwardIcon\\"><path d=\\"M4 12l1.41 1.41L11 7.83V20h2V7.83l5.58 5.59L20 12l-8-8-8 8z\\"></path></svg><span class=\\"MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root\\"></span></button></div></div><div class=\\"MuiDataGrid-menuIcon\\"><button class=\\"MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall MuiDataGrid-menuIconButton css-1pe4mpk-MuiButtonBase-root-MuiIconButton-root\\" tabindex=\\"-1\\" type=\\"button\\" aria-label=\\"Menu\\" title=\\"Menu\\" aria-haspopup=\\"true\\" aria-controls=\\"mui-3\\" id=\\"mui-4\\"><svg class=\\"MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root\\" focusable=\\"false\\" aria-hidden=\\"true\\" viewBox=\\"0 0 24 24\\" data-testid=\\"TripleDotsVerticalIcon\\"><path d=\\"M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z\\"></path></svg><span class=\\"MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root\\"></span></button></div></div><div class=\\"MuiDataGrid-columnSeparator MuiDataGrid-columnSeparator--sideRight\\" style=\\"min-height: 56px; opacity: 1;\\"><svg class=\\"MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiDataGrid-iconSeparator css-i4bv87-MuiSvgIcon-root\\" focusable=\\"false\\" aria-hidden=\\"true\\" viewBox=\\"0 0 24 24\\" data-testid=\\"SeparatorIcon\\"><path d=\\"M11 19V5h2v14z\\"></path></svg></div></div><div class=\\"MuiDataGrid-columnHeader MuiDataGrid-columnHeader--sortable\\" data-field=\\"edit\\" style=\\"width: 60px; min-width: 60px; max-width: 60px;\\" role=\\"columnheader\\" tabindex=\\"-1\\" aria-colindex=\\"2\\" aria-sort=\\"none\\"><div class=\\"MuiDataGrid-columnHeaderDraggableContainer\\" draggable=\\"false\\"><div class=\\"MuiDataGrid-columnHeaderTitleContainer\\"><div class=\\"MuiDataGrid-columnHeaderTitleContainerContent\\"><div class=\\"MuiDataGrid-columnHeaderTitle css-1jbbcbn-MuiDataGrid-columnHeaderTitle\\" aria-label=\\"\\" data-mui-internal-clone-element=\\"true\\">edit</div></div><div class=\\"MuiDataGrid-iconButtonContainer css-ltf0zy-MuiDataGrid-iconButtonContainer\\"><button class=\\"MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-1pe4mpk-MuiButtonBase-root-MuiIconButton-root\\" tabindex=\\"-1\\" type=\\"button\\" aria-label=\\"Sort\\" title=\\"Sort\\"><svg class=\\"MuiSvgIcon-root MuiSvgIcon-fontSizeSmall MuiDataGrid-sortIcon css-ptiqhd-MuiSvgIcon-root\\" focusable=\\"false\\" aria-hidden=\\"true\\" viewBox=\\"0 0 24 24\\" data-testid=\\"ArrowUpwardIcon\\"><path d=\\"M4 12l1.41 1.41L11 7.83V20h2V7.83l5.58 5.59L20 12l-8-8-8 8z\\"></path></svg><span class=\\"MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root\\"></span></button></div></div><div class=\\"MuiDataGrid-menuIcon\\"><button class=\\"MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall MuiDataGrid-menuIconButton css-1pe4mpk-MuiButtonBase-root-MuiIconButton-root\\" tabindex=\\"-1\\" type=\\"button\\" aria-label=\\"Menu\\" title=\\"Menu\\" aria-haspopup=\\"true\\" aria-controls=\\"mui-6\\" id=\\"mui-7\\"><svg class=\\"MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root\\" focusable=\\"false\\" aria-hidden=\\"true\\" viewBox=\\"0 0 24 24\\" data-testid=\\"TripleDotsVerticalIcon\\"><path d=\\"M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z\\"></path></svg><span class=\\"MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root\\"></span></button></div></div><div class=\\"MuiDataGrid-columnSeparator MuiDataGrid-columnSeparator--sideRight\\" style=\\"min-height: 56px; opacity: 1;\\"><svg class=\\"MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiDataGrid-iconSeparator css-i4bv87-MuiSvgIcon-root\\" focusable=\\"false\\" aria-hidden=\\"true\\" viewBox=\\"0 0 24 24\\" data-testid=\\"SeparatorIcon\\"><path d=\\"M11 19V5h2v14z\\"></path></svg></div></div></div></div><div style=\\"overflow: visible; width: 0px;\\"><div class=\\"MuiDataGrid-virtualScroller css-1w5m2wr-MuiDataGrid-virtualScroller\\" style=\\"width: 0px; height: auto; margin-top: 56px; overflow-x: hidden; overflow-y: hidden;\\"><div class=\\"MuiDataGrid-virtualScrollerContent css-1kwdphh-MuiDataGrid-virtualScrollerContent\\" style=\\"width: auto; height: 156px; min-height: auto;\\"><div class=\\"MuiDataGrid-virtualScrollerRenderZone css-s1v7zr-MuiDataGrid-virtualScrollerRenderZone\\" style=\\"transform: translate3d(0px, 0px, 0px);\\"><div data-id=\\"1\\" data-rowindex=\\"0\\" role=\\"row\\" class=\\"MuiDataGrid-row\\" aria-rowindex=\\"2\\" aria-selected=\\"false\\" style=\\"max-height: 52px; min-height: 52px;\\"><div class=\\"MuiDataGrid-cell MuiDataGrid-cell--textLeft\\" role=\\"cell\\" data-field=\\"id\\" data-colindex=\\"0\\" aria-colindex=\\"1\\" aria-colspan=\\"1\\" style=\\"min-width: 90px; max-width: 90px; min-height: 52px; max-height: 52px;\\" tabindex=\\"-1\\"><div class=\\"MuiDataGrid-cellContent\\">1</div></div><div class=\\"MuiDataGrid-cell--withRenderer MuiDataGrid-cell MuiDataGrid-cell--textLeft MuiDataGrid-cell--editable\\" role=\\"cell\\" data-field=\\"edit\\" data-colindex=\\"1\\" aria-colindex=\\"2\\" aria-colspan=\\"1\\" style=\\"min-width: 60px; max-width: 60px; min-height: 52px; max-height: 52px;\\" tabindex=\\"-1\\"><button class=\\"MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium mockButton css-78trlr-MuiButtonBase-root-MuiIconButton-root\\" tabindex=\\"0\\" type=\\"button\\"><svg class=\\"MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root\\" focusable=\\"false\\" aria-hidden=\\"true\\" viewBox=\\"0 0 24 24\\" data-testid=\\"RemoveRedEyeIcon\\"><path d=\\"M12 4.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5zM12 17c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5zm0-8c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z\\"></path></svg><span class=\\"MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root\\"></span></button></div></div><div data-id=\\"2\\" data-rowindex=\\"1\\" role=\\"row\\" class=\\"MuiDataGrid-row\\" aria-rowindex=\\"3\\" aria-selected=\\"false\\" style=\\"max-height: 52px; min-height: 52px;\\"><div class=\\"MuiDataGrid-cell MuiDataGrid-cell--textLeft\\" role=\\"cell\\" data-field=\\"id\\" data-colindex=\\"0\\" aria-colindex=\\"1\\" aria-colspan=\\"1\\" style=\\"min-width: 90px; max-width: 90px; min-height: 52px; max-height: 52px;\\" tabindex=\\"-1\\"><div class=\\"MuiDataGrid-cellContent\\">2</div></div><div class=\\"MuiDataGrid-cell--withRenderer MuiDataGrid-cell MuiDataGrid-cell--textLeft MuiDataGrid-cell--editable\\" role=\\"cell\\" data-field=\\"edit\\" data-colindex=\\"1\\" aria-colindex=\\"2\\" aria-colspan=\\"1\\" style=\\"min-width: 60px; max-width: 60px; min-height: 52px; max-height: 52px;\\" tabindex=\\"-1\\"><button class=\\"MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium mockButton css-78trlr-MuiButtonBase-root-MuiIconButton-root\\" tabindex=\\"0\\" type=\\"button\\"><svg class=\\"MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root\\" focusable=\\"false\\" aria-hidden=\\"true\\" viewBox=\\"0 0 24 24\\" data-testid=\\"RemoveRedEyeIcon\\"><path d=\\"M12 4.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5zM12 17c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5zm0-8c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z\\"></path></svg><span class=\\"MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root\\"></span></button></div></div><div data-id=\\"3\\" data-rowindex=\\"2\\" role=\\"row\\" class=\\"MuiDataGrid-row MuiDataGrid-row--lastVisible\\" aria-rowindex=\\"4\\" aria-selected=\\"false\\" style=\\"max-height: 52px; min-height: 52px;\\"><div class=\\"MuiDataGrid-cell MuiDataGrid-cell--textLeft\\" role=\\"cell\\" data-field=\\"id\\" data-colindex=\\"0\\" aria-colindex=\\"1\\" aria-colspan=\\"1\\" style=\\"min-width: 90px; max-width: 90px; min-height: 52px; max-height: 52px;\\" tabindex=\\"-1\\"><div class=\\"MuiDataGrid-cellContent\\">3</div></div><div class=\\"MuiDataGrid-cell--withRenderer MuiDataGrid-cell MuiDataGrid-cell--textLeft MuiDataGrid-cell--editable\\" role=\\"cell\\" data-field=\\"edit\\" data-colindex=\\"1\\" aria-colindex=\\"2\\" aria-colspan=\\"1\\" style=\\"min-width: 60px; max-width: 60px; min-height: 52px; max-height: 52px;\\" tabindex=\\"-1\\"><button class=\\"MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium mockButton css-78trlr-MuiButtonBase-root-MuiIconButton-root\\" tabindex=\\"0\\" type=\\"button\\"><svg class=\\"MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root\\" focusable=\\"false\\" aria-hidden=\\"true\\" viewBox=\\"0 0 24 24\\" data-testid=\\"RemoveRedEyeIcon\\"><path d=\\"M12 4.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5zM12 17c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5zm0-8c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z\\"></path></svg><span class=\\"MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root\\"></span></button></div></div></div></div></div></div><div class=\\"Mui-resizeTriggers\\"><div class=\\"expand-trigger\\"><div style=\\"width: 1px; height: 1px;\\"></div></div><div class=\\"contract-trigger\\"></div></div></div><div></div></div></div>"`;
